### PR TITLE
Simplify product tests JVM memory configuration

### PR DESCRIPTION
**Pull request by nineinchnick**
## Description

A follow-up to #16231 that reverts #3074.

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Review by electrum — APPROVED** [review]
Please compare the results, then I'll merge this if it's not a regression.

**Comment by nineinchnick** [comment]
Here's a highly scientific comparison of [this branch](https://github.com/trinodb/trino/actions/runs/4294555489) with [the latest green master run](https://github.com/trinodb/trino/actions/runs/4291913390). I included all tests present on master that are missing here just to show off how effective GIB is.

tl;dr there are differences up to 30%, both ways, so I don't think this is a safe change.

Good news is we can reuse this query to easily compare more runs.

```
                           job_name                            | diff_pct | master_job_duration | branch_job_duration |  duration_diff  
---------------------------------------------------------------+----------+---------------------+---------------------+-----------------
 error-prone-checks                                            |    46.33 | 0 00:22:01.000      | 0 00:11:49.000      | 0 00:10:12.000  
 web-ui-checks                                                 |      0.0 | 0 00:00:52.000      | 0 00:00:52.000      | 0 00:00:00.000  
 check-commits-dispatcher                                      |   -100.0 | 0 00:00:00.000      | 0 00:00:38.000      | -0 00:00:38.000 
 hive-tests (config-hdp3)                                      |    91.78 | 0 00:40:09.000      | 0 00:03:18.000      | 0 00:36:51.000  
 maven-checks (17)                                             |    -9.59 | 0 00:31:35.000      | 0 00:34:56.000      | -0 00:03:21.000 
 test-jdbc-compatibility                                       |    77.46 | 0 00:22:33.000      | 0 00:05:05.000      | 0 00:17:28.000  
 test-other-modules                                            |    81.17 | 0 00:28:25.000      | 0 00:05:21.000      | 0 00:23:04.000  
 build-test-matrix                                             |    -5.03 | 0 00:02:31.000      | 0 00:02:39.000      | -0 00:00:08.000 
 build-pt                                                      |     1.75 | 0 00:18:08.000      | 0 00:17:49.000      | 0 00:00:19.000  
 check-commit                                                  |     NULL | 0 00:00:00.000      | 0 00:00:00.000      | 0 00:00:00.000  
 test (client/trino-jdbc,plugin ... hrift,plugin/trino-memory) |     NULL | 0 00:15:31.000      | NULL                | NULL            
 test (core/trino-main)                                        |     NULL | 0 00:25:32.000      | NULL                | NULL            
 test (core/trino-main, 19)                                    |     NULL | 0 00:24:47.000      | NULL                | NULL            
 test (plugin/trino-accumulo)                                  |     NULL | 0 00:12:20.000      | NULL                | NULL            
 test (plugin/trino-bigquery)                                  |     NULL | 0 00:37:28.000      | NULL                | NULL            
 test (plugin/trino-bigquery, cloud-tests-arrow)               |     NULL | 0 00:29:21.000      | NULL                | NULL            
 test (plugin/trino-cassandra)                                 |     NULL | 0 00:19:30.000      | NULL                | NULL            
 test (plugin/trino-clickhouse)                                |     NULL | 0 00:13:34.000      | NULL                | NULL            
 test (plugin/trino-delta-lake)                                |     NULL | 0 00:33:36.000      | NULL                | NULL            
 test (plugin/trino-delta-lake, cloud-tests)                   |     NULL | 0 00:22:16.000      | NULL                | NULL            
 test (plugin/trino-delta-lake, gcs-tests)                     |     NULL | 0 00:23:01.000      | NULL                | NULL            
 test (plugin/trino-druid)                                     |     NULL | 0 00:45:20.000      | NULL                | NULL            
 test (plugin/trino-elasticsearch)                             |     NULL | 0 00:14:15.000      | NULL                | NULL            
 test (plugin/trino-hive)                                      |     NULL | 0 00:38:41.000      | NULL                | NULL            
 test (plugin/trino-hive, test-parquet)                        |     NULL | 0 00:32:56.000      | NULL                | NULL            
 test (plugin/trino-hudi)                                      |     NULL | 0 00:16:26.000      | NULL                | NULL            
 test (plugin/trino-iceberg)                                   |     NULL | 0 00:53:03.000      | NULL                | NULL            
 test (plugin/trino-iceberg, additional-catalog-tests)         |     NULL | 0 00:24:50.000      | NULL                | NULL            
 test (plugin/trino-iceberg, cloud-tests)                      |     NULL | 0 00:15:48.000      | NULL                | NULL            
 test (plugin/trino-ignite)                                    |     NULL | 0 00:10:56.000      | NULL                | NULL            
 test (plugin/trino-kafka)                                     |     NULL | 0 00:16:28.000      | NULL                | NULL            
 test (plugin/trino-kudu)                                      |     NULL | 0 00:16:07.000      | NULL                | NULL            
 test (plugin/trino-mariadb)                                   |     NULL | 0 00:12:43.000      | NULL                | NULL            
 test (plugin/trino-mongodb)                                   |     NULL | 0 00:10:19.000      | NULL                | NULL            
 test (plugin/trino-mysql)                                     |     NULL | 0 00:14:25.000      | NULL                | NULL            
 test (plugin/trino-oracle)                                    |     NULL | 0 00:16:18.000      | NULL                | NULL            
 test (plugin/trino-phoenix5)                                  |     NULL | 0 00:19:58.000      | NULL                | NULL            
 test (plugin/trino-pinot)                                     |     NULL | 0 00:23:38.000      | NULL                | NULL            
 test (plugin/trino-postgresql)                                |     NULL | 0 00:16:33.000      | NULL                | NULL            
 test (plugin/trino-raptor-legacy)                             |     NULL | 0 00:18:26.000      | NULL                | NULL            
 test (plugin/trino-redis)                                     |     NULL | 0 00:09:11.000      | NULL                | NULL            
 test (plugin/trino-singlestore)                               |     NULL | 0 00:17:42.000      | NULL                | NULL            
 test (plugin/trino-sqlserver)                                 |     NULL | 0 00:18:25.000      | NULL                | NULL            
 test (testing/trino-faulttolerant-tests, default)             |     NULL | 0 00:09:28.000      | NULL                | NULL            
 test (plugin/trino-delta-lake, fte-tests)                     |     NULL | 0 00:12:04.000      | NULL                | NULL            
 test (testing/trino-faulttoler ... test-fault-tolerant-delta) |     NULL | 0 00:08:32.000      | NULL                | NULL            
 test (plugin/trino-hive, fte-tests)                           |     NULL | 0 00:23:18.000      | NULL                | NULL            
 test (testing/trino-faulttoler ...  test-fault-tolerant-hive) |     NULL | 0 00:27:10.000      | NULL                | NULL            
 test (plugin/trino-iceberg, fte-tests)                        |     NULL | 0 00:15:33.000      | NULL                | NULL            
 test (testing/trino-faulttoler ... st-fault-tolerant-iceberg) |     NULL | 0 00:16:31.000      | NULL                | NULL            
 test (plugin/trino-postgresql, fte-tests)                     |     NULL | 0 00:08:38.000      | NULL                | NULL            
 test (plugin/trino-mongodb, fte-tests)                        |     NULL | 0 00:08:36.000      | NULL                | NULL            
 test (plugin/trino-mysql, fte-tests)                          |     NULL | 0 00:09:57.000      | NULL                | NULL            
 test (plugin/trino-sqlserver, fte-tests)                      |     NULL | 0 00:13:32.000      | NULL                | NULL            
 test (testing/trino-tests)                                    |     NULL | 0 00:35:40.000      | NULL                | NULL            
 pt (default, suite-1, )                                       |     NULL | 0 00:56:44.000      | NULL                | NULL            
 pt (default, suite-2, )                                       |     NULL | 0 00:38:45.000      | NULL                | NULL            
 pt (default, suite-3, )                                       |     NULL | 0 00:39:33.000      | NULL                | NULL            
 pt (default, suite-5, )                                       |     NULL | 0 00:36:54.000      | NULL                | NULL            
 pt (default, suite-delta-lake-databricks73, )                 |     NULL | 0 00:28:31.000      | NULL                | NULL            
 pt (default, suite-delta-lake-databricks91, )                 |     NULL | 0 00:58:18.000      | NULL                | NULL            
 pt (default, suite-delta-lake-databricks104, )                |     NULL | 0 00:56:15.000      | NULL                | NULL            
 pt (default, suite-delta-lake-databricks113, )                |     NULL | 0 00:59:14.000      | NULL                | NULL            
 pt (default, suite-clients, )                                 |     NULL | 0 00:11:51.000      | NULL                | NULL            
 pt (default, suite-functions, )                               |     NULL | 0 00:04:35.000      | NULL                | NULL            
 pt (default, suite-tpch, )                                    |     NULL | 0 00:10:46.000      | NULL                | NULL            
 pt (default, suite-storage-formats-detailed, )                |     NULL | 0 00:59:54.000      | NULL                | NULL            
 pt (hdp3, suite-1, )                                          |   -13.11 | 0 00:52:27.000      | 0 01:00:22.000      | -0 00:07:55.000 
 pt (hdp3, suite-2, )                                          |    30.86 | 0 01:15:27.000      | 0 00:52:10.000      | 0 00:23:17.000  
 pt (hdp3, suite-3, )                                          |    -2.19 | 0 00:38:40.000      | 0 00:39:32.000      | -0 00:00:52.000 
 pt (hdp3, suite-5, )                                          |     0.86 | 0 00:56:01.000      | 0 00:55:32.000      | 0 00:00:29.000  
 pt (hdp3, suite-azure, )                                      |     NULL | 0 00:05:32.000      | NULL                | NULL            
 pt (hdp3, suite-gcs, )                                        |     NULL | 0 00:04:51.000      | NULL                | NULL            
 pt (hdp3, suite-clients, )                                    |    14.86 | 0 00:09:12.000      | 0 00:07:50.000      | 0 00:01:22.000  
 pt (hdp3, suite-functions, )                                  |    18.44 | 0 00:05:47.000      | 0 00:04:43.000      | 0 00:01:04.000  
 pt (hdp3, suite-tpch, )                                       |   -24.84 | 0 00:11:30.000      | 0 00:15:18.000      | -0 00:03:48.000 
 pt (hdp3, suite-storage-formats-detailed, )                   |    -2.71 | 0 00:18:32.000      | 0 00:19:03.000      | -0 00:00:31.000 
 pt (default, suite-6-non-generic, )                           |    -8.57 | 0 00:26:40.000      | 0 00:29:10.000      | -0 00:02:30.000 
 pt (default, suite-7-non-generic, )                           |   -28.37 | 0 00:26:23.000      | 0 00:36:50.000      | -0 00:10:27.000 
 pt (default, suite-8-non-generic, )                           |    20.36 | 0 00:44:07.000      | 0 00:35:08.000      | 0 00:08:59.000  
 pt (default, suite-tpcds, )                                   |     2.54 | 0 00:24:54.000      | 0 00:24:16.000      | 0 00:00:38.000  
 pt (default, suite-parquet, )                                 |     0.88 | 0 00:17:06.000      | 0 00:16:57.000      | 0 00:00:09.000  
 pt (default, suite-oauth2, )                                  |    15.82 | 0 00:12:07.000      | 0 00:10:12.000      | 0 00:01:55.000  
 pt (default, suite-ldap, )                                    |    -2.58 | 0 00:12:35.000      | 0 00:12:55.000      | -0 00:00:20.000 
 pt (default, suite-compatibility, )                           |    -2.74 | 0 00:23:38.000      | 0 00:24:18.000      | -0 00:00:40.000 
 pt (apache-hive3, suite-hms-only, )                           |   -14.64 | 0 00:07:58.000      | 0 00:09:20.000      | -0 00:01:22.000 
 pt (default, suite-all, )                                     |     8.91 | 0 00:04:07.000      | 0 00:03:45.000      | 0 00:00:22.000  
 pt (default, suite-delta-lake-oss, )                          |     8.99 | 0 00:20:01.000      | 0 00:18:13.000      | 0 00:01:48.000  
 pt (default, suite-kafka, )                                   |   -23.15 | 0 00:08:08.000      | 0 00:10:35.000      | -0 00:02:27.000 
 pt (default, suite-cassandra, )                               |      3.2 | 0 00:05:44.000      | 0 00:05:33.000      | 0 00:00:11.000  
 pt (default, suite-clickhouse, )                              |      6.9 | 0 00:02:54.000      | 0 00:02:42.000      | 0 00:00:12.000  
 pt (default, suite-mysql, )                                   |    -7.92 | 0 00:04:04.000      | 0 00:04:25.000      | -0 00:00:21.000 
 pt (default, suite-iceberg, )                                 |     4.88 | 0 00:57:00.000      | 0 00:54:13.000      | 0 00:02:47.000  
 pt (default, suite-hudi, )                                    |    -1.56 | 0 00:13:38.000      | 0 00:13:51.000      | -0 00:00:13.000 
 pt (default, suite-ignite, )                                  |    -2.39 | 0 00:03:24.000      | 0 00:03:29.000      | -0 00:00:05.000 
 test                                                          |     NULL | NULL                | 0 00:00:00.000      | NULL            
(96 rows)
```

The query:
```sql
WITH
jobs AS (
    SELECT
        run_id
      , name
      , completed_at - started_at AS duration
      , case when length(name) > 60 then substr(name, 1, 30) || ' ... ' || substr(name, length(name) - 25) else name end AS short_name
    FROM jobs
)
, master_jobs AS (
    SELECT *
    FROM jobs
    WHERE run_id = 4291913390
)
, branch_jobs AS (
    SELECT *
    FROM jobs
    WHERE run_id = 4294555489
)
SELECT
    coalesce(mj.short_name, bj.short_name) AS job_name
  , round(100 * to_milliseconds(mj.duration - bj.duration) / cast(nullif(to_milliseconds(greatest(mj.duration, bj.duration)), 0) as double), 2) AS diff_pct
  , mj.duration AS master_job_duration
  , bj.duration AS branch_job_duration
  , mj.duration - bj.duration AS duration_diff
FROM master_jobs mj
FULL OUTER JOIN branch_jobs bj ON bj.name = mj.name
;

```
